### PR TITLE
Automate update of consistency test baseline data.

### LINKED ISSUE #574: Automate update of consistency test baseline data
Code updates often change results, which means the consistency test baseline data must be updated. Currently, updates are done manually on five machines. But as the number of weekly PRs grows, we need an automated way to do this.

### PR DESCRIPTION
**Pull request by GeorgeGayno-NOAA**
## DESCRIPTION OF CHANGES: 
Add logic to automatically update the consistency test baseline data when a test fails. 

## TESTS CONDUCTED: 
Tested on Hera and Dell using a sub-set of the total test suite.

## DEPENDENCIES:
None.

## DOCUMENTATION:
N/A

## ISSUE: 
Fixes #574

### CODE REVIEW

**Comment by GeorgeGayno-NOAA** [comment]
@LarissaReames-NOAA and @JeffBeck-NOAA Would appreciate your comments before I update the entire test suite. You can try this on Hera or Dell.  Copy the consistency test data directories to your own space, then update HOMEreg in the driver script. To get a test to fail, you can change the link to point to an old hash of the baseline data:

```
lrwxrwxrwx 1 George.Gayno emcmodel   28 Sep  1 17:39 25km_conus_gfs_grib2 -> 25km_conus_gfs_grib2.2ad0033
dr-xr-xr-x 2 George.Gayno g01      2048 Aug 25 20:25 25km_conus_gfs_grib2.2ad0033
dr-xr-xr-x 2 George.Gayno g01      2048 Aug 13 12:47 25km_conus_gfs_grib2.469fd5b
dr-xr-sr-x 2 George.Gayno emcmodel 2048 Jun  7 15:57 25km_conus_gfs_grib2.529b168
```

File: reg_tests/update_baseline.sh [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  if [ -d $base_dir_commit/fix_sfc ]; then
+    chmod 777 $base_dir_commit/fix_sfc
+  fi
+  rm -fr $base_dir_commit
+fi
+
+mkdir -p $base_dir_commit
+
+for files in *.nc
+do
```
Comment (JeffBeck-NOAA) [inline reg_tests/update_baseline.sh:25]: I see there's no 'cd' here before NetCDF files are copied. Is the `reg_tests` directory defined to be the location of all the RT NetCDF files somewhere?  Will this work for for all utilities?

File: reg_tests/chgres_cube/driver.hera.sh [hunk 2]
```diff
@@ -44,9 +44,14 @@ QUEUE="${QUEUE:-batch}"
 # and baseline data for each test.
 #-----------------------------------------------------------------------------
 
+#export UPDATE_BASELINE="FALSE"
+export UPDATE_BASELINE="TRUE"
+
+source ../get_hash.sh
+
```
Comment (JeffBeck-NOAA) [inline reg_tests/chgres_cube/driver.hera.sh:53]: Should this only be sourced if UPDATE_BASELINE="TRUE"?

**Review by JeffBeck-NOAA — COMMENTED** [review]
This code looks good.  I just had a couple questions related to clarification.  Thanks!

File: reg_tests/chgres_cube/driver.hera.sh (hunk 2 above)
Comment (JeffBeck-NOAA) [reply reg_tests/chgres_cube/driver.hera.sh:53]: Do we manually set UPDATE_BASELINE="TRUE" and rerun the RTs when we know we want to replace the baselines?

Comment (GeorgeGayno-NOAA) [reply reg_tests/chgres_cube/driver.hera.sh:53]: > Should this only be sourced if UPDATE_BASELINE="TRUE"?

Good suggestion. I will make that change.

Comment (GeorgeGayno-NOAA) [reply reg_tests/chgres_cube/driver.hera.sh:53]: > Do we manually set UPDATE_BASELINE="TRUE" and rerun the RTs when we know we want to replace the baselines?

Yes. After a merge that changes results, the corresponding scripts would be run with UPDATE_BASELINE="TRUE" on all machines. This should only have to be done once after a merge.

File: reg_tests/update_baseline.sh (hunk 1 above)
Comment (GeorgeGayno-NOAA) [reply reg_tests/update_baseline.sh:25]: This script is run from the test's working directory. So the files are copied from the working directory to the baseline directory. The baseline directory for the particular test is stored in `base_dir.`  And `base_dir` is a function of `HOMEreg`, which is the regression test root directory and is passed in from the driver script.

File: reg_tests/chgres_cube/driver.hera.sh (hunk 2 above)
Comment (JeffBeck-NOAA) [reply reg_tests/chgres_cube/driver.hera.sh:53]: Sounds good!

File: reg_tests/chgres_cube/driver.hera.sh [hunk 3]
```diff
@@ -44,9 +44,16 @@ QUEUE="${QUEUE:-batch}"
...
+#export UPDATE_BASELINE="TRUE"
+
+if [ "$UPDATE_BASELINE" = "TRUE" ]; then
+  source ../get_hash.sh
+fi
+
 export HOMEufs=$PWD/../..
 
-export HOMEreg=/scratch1/NCEPDEV/nems/role.ufsutils/ufs_utils/reg_tests/chgres_cube
+export HOMEreg=/scratch1/NCEPDEV/da/George.Gayno/noscrub/reg_tests/chgres_cube.test
```
Comment (JeffBeck-NOAA) [inline reg_tests/chgres_cube/driver.hera.sh]: I assume the personal HOMEreg paths will get reset to their original values before merging?

**Review by JeffBeck-NOAA — APPROVED** [review]

File: reg_tests/chgres_cube/driver.hera.sh (hunk 3 above)
Comment (GeorgeGayno-NOAA) [reply reg_tests/chgres_cube/driver.hera.sh]: Yes. That is just for testing.

**Comment by GeorgeGayno-NOAA** [comment]
As a final check, I tested my branch on Hera, Jet and Orion using the role account. It worked. Will merge soon.

**Pull request merged**